### PR TITLE
[SPARK-4858] Add an option to turn off a progress bar in spark-shell

### PR DESCRIPTION
**Pull request by maropu**
Add an '--no-progress-bar' option to easily turn off a progress bar in spark-shell for users who'd like to look into debug logs or something.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala [hunk 1]
```diff
@@ -380,6 +386,8 @@ private[spark] class SparkSubmitArguments(args: Seq[String], env: Map[String, St
         |
         |  --executor-memory MEM       Memory per executor (e.g. 1000M, 2G) (Default: 1G).
         |
+        |  --no-progress-bar           Turn off a console progress bar
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala]: Turn off the

File: bin/utils.sh [hunk 2]
```diff
@@ -32,10 +32,11 @@ function gatherSparkSubmitOpts() {
   APPLICATION_OPTS=()
   while (($#)); do
     case "$1" in
-      --master | --deploy-mode | --class | --name | --jars | --py-files | --files | \
-      --conf | --properties-file | --driver-memory | --driver-java-options | \
-      --driver-library-path | --driver-class-path | --executor-memory | --driver-cores | \
-      --total-executor-cores | --executor-cores | --queue | --num-executors | --archives)
+      --master | --deploy-mode | --class | --name | --jars | --no-progress-bar | \
```
Comment (andrewor14) [inline bin/utils.sh]: please also update `bin/windows-utils.cmd`

**Comment by andrewor14** [comment]
@davies any other comments?

File: bin/utils.sh (hunk 2 above)
Comment (davies) [inline bin/utils.sh]: The diff will be minimized, easy to review and blame (in future).

**Comment by pwendell** [comment]
Just wondering, it seems like a fairly esoteric option for adding a built-in flag.

Why not just do this?
```
./bin/spark-shell --conf spark.ui.showConsoleProgress=false
```

Is it a common thing to disable this progress bar?

**Comment by davies** [comment]
@pwendell This is a better solution, we can not continue to add more flags for every useful user-cases.

**Comment by andrewor14** [comment]
Yeah that's a good point. It seems that `--conf` is sufficient for this use case. @maropu would you mind closing this issue?

**Comment by maropu** [comment]
Ok.

**Pull request closed without merging**